frontend/public/module/k8s/k8s-models.ts

```diff
@@ -72,3 +75,56 @@ export const modelFor = (ref: K8sResourceKindReference) => {
  * NOTE: This will not work for CRDs defined at runtime, use `connectToModels` instead.
  */
 export const allModels = () => k8sModels;
+
+/**
+ * Use this hook to find the model for resources using only group and kind
+ */
+export const useModelFinder = () => {
+  const models: K8sKind[] = _.values(
+    useSelector(({ k8s }) => k8s.getIn(['RESOURCES', 'models'])).toJS(),
+  );
+  const groupVersionMap: DiscoveryResources['groupVersionMap'] = useSelector(({ k8s }) =>
+    k8s.getIn(['RESOURCES', 'groupToVersionMap']),
+  );
+
+  const findModel = (group: string, kind: string) => {
+    const groupObject = _.find(groupVersionMap, (item) => item.name === group);
+    const { preferredVersion, versions } = groupObject || {};
+    let model = undefined;
+    if (preferredVersion) {
+      // Find a model for the CRD that uses this preferred version
+      model = _.find(
+        models,
+        (item) =>
+          item.apiGroup === group &&
+          item.apiVersion === preferredVersion &&
+          (item.kind === kind || item.plural === kind),
+      );
+    }
+    if (model) {
+      return model;
+    }
+    // In case the preferred version does not have the CRD
+    if (versions) {
+      const sortedVersions: string[] = versions.sort(apiVersionCompare);
+      sortedVersions.forEach((ver: string) => {
+        const model2 = _.find(
+          models,
+          (item) =>
+            item.apiGroup === group &&
+            item.apiVersion === ver &&
+            (item.kind === kind || item.plural === kind),
+        );
+        if (model2) {
+          return model2;
+        }
+      });
+    }
+    // If we have reached this point then it has no api group and the versions are all v1
```

Comment:
I'm not sure this comment is right. What if there was an API group passed, but it's one we didn't find during discovery. This has a chance of returning the wrong model.

We should handle that case first IMO and return early.

Reply:
We should reach here only if the group is empty and that should happen for K8s native kind. The workaround it would be put a check that group is indeed empty and then run it. If we can't find anything we can just give up and return null and we would render a linkless name in the list.

Reply:
> We should reach here only if the group is empty and that should happen for K8s native kind.

Wouldn't we get here also if the API group is one that isn't available? (Say a CRD isn't installed or there's just a typo.)

I'd handle the case where API group is missing or `''` first and return early.

Reply:
I agree with you. I realized it when making changes. The code is a bit different. Sorry for the confusion.